src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java

```diff
@@ -694,18 +700,66 @@ private void configureFullTextSearch() {
   }
 
   private FluentFuture configureFileDownload() {
-    FluentFuture<RestrictionInformationResponse> future = getDoneFuture(null);
+    FluentFuture<RestrictionInformationResponse> restrictionInformationFuture = getDoneFuture(
+      null
+    );
 
     if (entity instanceof FileEntity) {
       boolean canDownload = entityBundle.getPermissions().getCanDownload();
       actionMenu.setDownloadMenuEnabled(canDownload);
       if (canDownload) {
         actionMenu.setDownloadMenuTooltipText("");
       } else {
-        String viewOnlyHelpText = authenticationController.isLoggedIn()
-          ? "You don't have download permission. Request access from an administrator, shown under File Tools ➔ File Sharing Settings"
-          : "You need to log in to download this file.";
-        actionMenu.setDownloadMenuTooltipText(viewOnlyHelpText);
+        if (!authenticationController.isLoggedIn()) {
+          actionMenu.setDownloadMenuTooltipText(
+            "You need to log in to download this file."
+          );
+        } else {
+          actionMenu.setDownloadMenuTooltipText(
+            "You don't have permission to download this file."
+          );
+          // Queue up request to identify reasons why the file cannot be downloaded, and update the tooltip when the request finishes.
+          getSynapseJavascriptClient()
+            .getActionsRequiredForEntityDownload(entity.getId())
+            .addCallback(
+              new FutureCallback<ActionRequiredList>() {
+                @Override
+                public void onSuccess(@NullableDecl ActionRequiredList result) {
+                  if (result != null) {
+                    StringBuilder downloadMenuTooltipText = new StringBuilder(
+                      "You don't have permission to download this file."
+                    );
+                    // There may be multiple actions of the same class, but we only want to show one message for each type
+                    // Get the unique set of action classes.
+                    Set<Class<? extends org.sagebionetworks.repo.model.download.Action>> uniqueClasses = result
```

Comment:
Unfortunately there's another `Action` class in use here, so we have to use the full classname